packages/cli-kit/src/session/authorize.test.ts

```diff
@@ -57,4 +65,45 @@ describe('authorize', () => {
     // Then
     await expect(auth).rejects.toThrowError(MismatchStateError)
   })
+
+  it('throws cancel execution exception if the port used for listing for the authorization response is already in use and the user do not want to terminate the process', async () => {
+    // Given
+    vi.mocked(checkPort).mockResolvedValue(false)
+    vi.mocked(terminateBlockingPortProcessPrompt).mockResolvedValue(false)
+
+    // When
+    const auth = authorize(['scope1', 'scope2'], 'state')
+
+    // Then
+    await expect(auth).rejects.toThrowError(new CancelExecution())
+    expect(killPortProcess).toBeCalledTimes(0)
+  })
+
+  it('terminate process if the port used for listing for the authorization response is already in use and the user confirm to terminate the process', async () => {
+    // Given
+    const challenge = {
+      codeChallenge: 'challenge',
+      codeVerifier: 'verifier',
+    }
+    vi.mocked(checkPort).mockResolvedValue(port)
+    vi.mocked(randomHex).mockReturnValue('hex')
+    vi.mocked(generateRandomChallengePair).mockReturnValue(challenge)
+    vi.mocked(listenRedirect).mockResolvedValue({code: 'code', state: 'state'})
+    vi.mocked(identity).mockResolvedValue('fqdn.com')
+    vi.mocked(clientId).mockResolvedValue('clientId')
+    vi.mocked(checkPort).mockResolvedValue(false)
```

Comment:
you are mocking `checkPort` twice here, I guess this is not intended and only the bottom one will be applied?